### DISCUSSION TITLE
about pyswmm

### DISCUSSION THREAD

**Discussion by Gaga279** [pyswmm/pyswmm#442] [Q&A] [answered]
I am a beginner of the swmm model, and recently followed the tutorials on the official website. But the problem is that my code always gives an error and I don't know why. Hope to get some help, thanks!
<img width="1089" alt="截屏2023-05-31 16 53 09" src="https://github.com/pyswmm/pyswmm/assets/135097185/29db4dc7-b31b-4917-aa04-036b8db8edf8">

**Comment by bemcdonnell** [comment]
Hi @Gaga279. Learning a coding language is tough.  Based on what I am seeing, you’re attempting to use ipython notebook.  For an initial attempt at getting it to work, have you tried running python using IDLE? It’s The “onboard” Python development environment.  My concern with using IPYNB is that it might not be linked to version of python that you have your pyswmm features installed.

**Reply by Gaga279** [reply]
Thank you very much for your reply. I was very  surprised that I didn't expect such a small question to be answered. About this issue has been resolved. In future studies, I will try to run python using idle and work hard to learn  python. Once again express my thanks!

**Comment by karosc** [comment, marked as answer]
@Gaga279, sometimes the pylance warnings in VSCode are unwarranted, but they can also be useful. Have you actually tried running the code? Or did the warnings from pylance trigger you to reach out here?

To be able to use notebooks properly in VSCode, you'll need to do a few things.

1. Make sure you have the tutorial dependencies installed! 
	* `pip install pyswmm matplotlib` 
2. make sure you have ipykernel installed 
	* `pip install ipykernel`
3. Make sure you've selected the proper python interpreter inside the jupyter notebook in VSCode. Python allows you to create different programming environments to keep your projects' dependencies isolated from each other. This is a very GOOD idea! VSCode has great support for virtual environments and even multiple installations of python, so just make sure the python interpreter selected in VSCode is the one you used to installed the dependencies. Do some googling about how to setup a *python virtual environment* if you want to learn more about these things.

	* 	![image](https://github.com/pyswmm/pyswmm/assets/36245370/dabec872-8f55-4967-a61f-a5fa2e8e0e8f)

**Reply by Gaga279** [reply]
Thank you very much for your reply! After getting your hint, I did supplementary study on relevant content, and finally solved the problem. I will continue to learn about  how to setup a python virtual environment . Thank you!
